services/api/src/migrations/2-harbor/harborSync.ts

```diff
@@ -0,0 +1,42 @@
+import * as R from 'ramda';
+import { logger } from '@lagoon/commons/src/local-logging';
+import { getSqlClient } from '../../clients/sqlClient';
+import createHarborOperations from '../../resources/project/harborSetup';
+import projectHelpers from '../../resources/project/helpers';
+import { query, prepare } from '../../util/db';
+
+(async () => {
+  const sqlClient = getSqlClient();
+  const harborClient = createHarborOperations(sqlClient);
+
+  logger.info('Syncing Harbor projects with Lagoon projects');
+  // Get a list of all Lagoon projects
+  const projects = await projectHelpers(sqlClient).getAllProjects();
+  for(var i=0; i < projects.length; i++) {
+    var project = projects[i];
+
+    // Get project's env vars
+    var envVars = await query(sqlClient, `SELECT * FROM env_vars WHERE project='${project.id}';`);
+    var hasVars = false;
+    if (envVars != null) {
```

Comment:
Did you test this specific condition? Our sql library is weird and I don't remember if returning 0 results actually returns `null` or if the `!=` is coercing the actual result (I think an object?) to `null` and so it's true by accident.

Reply:
Good catch; this was an artifact from my earlier testing attempts. I removed this logic, and the script still functions as intended.